### ISSUE TITLE
archive.auto doesn't work with JupyterLab

### ISSUE THREAD

**Issue by sappjw** [pyviz/holoviews#3570]
It seems that `hv.archive.auto()` doesn't work with JupyterLab because it doesn't allow Javascript execution:

```python
    >>> hv.archive.auto()
    
    Javascript Error: IPython is not defined
    Automatic capture is now enabled. [2019-03-21 11:03:54]
    ...
    >>> hv.archive.export()
    
    Export name: '2019_03_21-11_03_54'
    Directory    '<snip>'

    If no output appears, please check holoviews.archive.last_export_status()
    Javascript Error: IPython is not defined
    
    hv.archive.last_export_status()
    
    Status of the last call to holoviews.archive.export is unknown.
    (Re-execute this method once kernel status is idle.)
```

Is there a way to work around this limitation for JupyterLab notebooks?

**Comment by jlstevens** [comment]
Thanks for reporting this!

Frankly, I am not surprised - the only way this ever worked in classic notebook was with some dubious JavaScript and as you point out JupyterLab doesn't allow JavaScript execution.

I would love to be able to fix this if I thought the fix were easy, but I suspect the effort required doesn't really justifiy it. Maybe the docs can mention this only works in classic notebook or maybe there is some easier way to make it work that I don't know about... @philippjfr any ideas?

**Comment by jlstevens** [comment]
Thinking about this more, it is likely that the only part that is tricky in jupyterlab is grabbing the notebook JSON state - not the export of the images etc. As a result, I can probably issue a warning when things fail which means it can work properly in classic notebook and it can *mostly* work in Jupyterlab.

**Comment by gioarma** [comment]
I have encountered this problem as well, is there any news about this? 
I think it's quite a shame that such an important feature does not work in JupyterLab, especially since I guess more and more people are switching from Jupyter notebooks to JupyterLab.

**Comment by philippjfr** [comment]
No, sadly not, in fact we are likely to remove this functionality for this reason.

**Comment by jbednar** [comment]
It would be great for this to work in JupyterLab; I'm betting on @jlstevens figuring out a way. We developed it about the same time as I stopped writing academic papers, but if I were still writing them I'd sure want to have this functionality.

**Comment by gioarma** [comment]
@jbednar Yes, for a person like me who deals with writing reports and academic papers this functionality would be a game changer. 
I understand you have a lot of stuff to maintain and it's not easy, but I hope you will be able to do it because this is key for making reproducible science easy.

**Comment by toddrjen** [comment]
I am still encountering this issue. Considering that jupyterlab is the primary way of using notebooks today, it would probably be good to mention this issue in the documentation.

**Comment by jlstevens** [comment]
@hoxbro I would be in favor of either removing this functionality entirely at this point or at minimum mentioning that it only supports classic notebook in the docs...

**Comment by hoxbro** [comment]
> [@hoxbro](https://github.com/hoxbro) I would be in favor of either removing this functionality entirely at this point or at minimum mentioning that it only supports classic notebook in the docs...

I trust your judgment. Can you open a PR with your suggestion, and then we can take it from there?
